File: ezbroWorld/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min = prices[0]
        profit = 0
        for pointer in range(1, len(prices)):
            if min > prices[pointer] : 
                min = prices[pointer]
            elif min <= prices[pointer] : 
                profit = max(profit,prices[pointer]-min)
                
        return profit

File: ezbroWorld/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0], maxProfit = 0; 
        for(int i =1 ; i<prices.length;i++){
            if(min > prices[i]){
                min = prices[i];
            }
            else{
                maxProfit = Math.max(maxProfit, prices[i]-min);
            }
        }
        return maxProfit;
    }
}
